src/simulation.jl

```diff
@@ -101,19 +109,28 @@ Base.Expr(c::AllocVecCall) = begin
     _ => throw(AllocVecCallError(c))
   end
 
+  hook_ExprAVC_generate_cache_expr(c, resolved_form, c.code_target)
+end
+
+function hook_ExprAVC_generate_cache_expr(c::AllocVecCall, resolved_form::Symbol, ::cpu)
   :($(Symbol(:__,c.name)) = Decapodes.FixedSizeDiffCache(Vector{$(c.T)}(undef, nparts(mesh, $(QuoteNode(resolved_form))))))
 end
 
-#= function get_form_number(d::SummationDecapode, var_id::Int)
-type = d[var_id, :type]
-if(type == :Form0)
-return 0
-elseif(type == :Form1)
-return 1
-elseif(type == :Form2)
-return 2
+# TODO: Allow user to overload these hooks with user-defined code_target
+function hook_ExprAVC_generate_cache_expr(c::AllocVecCall, resolved_form::Symbol, ::cuda)
```

Comment:
This name is too long

Reply:
Sure but this is not exposed to the user, is still descriptive and can be changed when the ```TODO``` is done.

Reply:
I've changed the hook names to be shorter, in the format ```hook_$(FunctionAcronymWhereCalled)_$(BehaviorBeingModified)```